uktrade/omis-frontend
# Upgrade webpack

## Description of changes

Upgrading from Webpack 3 to Webpack 5, as well as some other related changes:
- `babel-preset-es2015` has been replaced by `@babel/preset-env`
- The other remaining Babel packages have been upgraded to v7
- `extract-text-webpack-plugin` has been deprecated, so it has been replaced by `mini-css-extract-plugin`
- `node-sass` has been replaced by `sass`
- `fast-sass-loader` is no longer being maintained, so it has been replaced by `sass-loader`
- `file-loader` has been deprecated and has been replace by native Webpack functionality
- All remaining webpack dependencies and loaders are on the latest version
- The security overrides have been cleared out as many of them were only appearing because of the old dependencies

## Test instructions
This branch has been deployed to the OMIS dev environment and tested there. In order to test out the new infrastructure you'll need to do the following:
- Find an OMIS order in dev. You'll need to enter enough information to get it to the 'quote awaiting acceptance' stage
- Once this is done click on 'Preview quote' and submit it
- Find the order in Django Admin and click on the link in the 'public facing URL' field (if you try and do this before the quote is submitted you will see an error page). This will take you to the order's page within the microservice itself to be tested.
